### PR TITLE
update mpas-source submodule

### PR DESCRIPTION
**Pull request by mark-petersen**
This update to mpas-source includes:

1. ocean: Update BGC commit hash
2. sea ice: Correction to prevent thermo energy conservation error
3. ocean: fix mixed layer depth for land ice
4. ocean: add variable-resolution GM Bolus capability

(4) adds five new ocean config flags, but default behavior for GM remains the same.  

[NML]
[non-BFB]

### CODE REVIEW

**Comment by mark-petersen** [comment]
Testing with:
```
SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.edison_gnu
PEM_Ln9.T62_oQU240.GMPAS-IAF.edison_gnu
PEM_Ln9.T62_oQU240.GMPAS-IAF.cori-haswell_gnu
PET_Ln9.T62_oEC60to30v3.GMPAS-IAF.cori-knl_gnu
PET_Ln9.T62_oEC60to30v3.GMPAS-IAF.cori-knl_intel
```

**Comment by jonbob** [comment]
@mark-petersen - the seaice change is not bfb because it changes a convergence criteria

**Comment by rljacob** [comment]
Does the BGC group need this on maint-1.1?

**Comment by jonbob** [comment]
@rljacob - the first two are already on maint-1.1, and I don't believe the second two are needed.

**Comment by maltrud** [comment]
i agree with @jonbob :  we don't want the second two.

File: components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml [hunk 1]
```diff
@@ -593,6 +593,14 @@ Valid values: Any positive real value, typically between 100 and 500 m.
 Default: Defined in namelist_defaults.xml
 </entry>
 
+<entry id="config_GM_Bolus_kappa_function" type="char*1024"
+	category="mesoscale_eddy_parameterization" group="mesoscale_eddy_parameterization">
+Control what type of function is used for GM Bolus kappa
+
```
Comment (jonbob) [inline components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml]: @mark-petersen - you have $\kappa$ in most places, but not here on 598?

File: components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml [hunk 2]
```diff
@@ -601,6 +609,44 @@ Valid values: Any positive real value.
...
+$\kappa$, max value in ramp as a function of dcEdge. Used when
+config_GM_Bolus_kappa_function is set to ramp.
+
+Valid values: Any positive real value.
+Default: Defined in namelist_defaults.xml
+</entry>
+
+<entry id="config_GM_Bolus_cell_size_min" type="real"
+	category="mesoscale_eddy_parameterization" group="mesoscale_eddy_parameterization">
+Minimum value in grid cell size for GM $kappa$ ramp function.  Here cell size
```
Comment (jonbob) [inline components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml]: do you want $\kappa$? it's missing the \

File: components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml [hunk 3]
```diff
@@ -601,6 +609,44 @@ Valid values: Any positive real value.
...
+refers to dcEdge. Used when config_GM_Bolus_kappa_function is set to ramp.
+
+Valid values: Any positive real value.
+Default: Defined in namelist_defaults.xml
+</entry>
+
+<entry id="config_GM_Bolus_cell_size_max" type="real"
+	category="mesoscale_eddy_parameterization" group="mesoscale_eddy_parameterization">
+Maximum value in grid cell size for GM $kappa$ ramp function.  Here cell size
```
Comment (jonbob) [inline components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml]: missing the \ again?

**Comment by jonbob** [comment]
@mark-petersen - thanks for fixing the xml file, but I think you didn't make the corresponding corrections to the Registry -- and that will cause problems running the scripts that auto-generate the build xml files

**Comment by mark-petersen** [comment]
Done.  Thanks for pointing that out.

**Comment by jonbob** [comment]
merged to next

**Comment by jonbob** [comment]
merged to master

**Pull request merged**